aws/resource_aws_dx_gateway_association.go

```diff
@@ -111,24 +189,66 @@ func resourceAwsDxGatewayAssociationRead(d *schema.ResourceData, meta interface{
 	}
 
 	assoc := assocRaw.(*directconnect.GatewayAssociation)
-	d.Set("dx_gateway_id", assoc.DirectConnectGatewayId)
-	d.Set("vpn_gateway_id", assoc.VirtualGatewayId)
-	d.Set("dx_gateway_association_id", assoc.AssociationId)
+
 	err = d.Set("allowed_prefixes", flattenDxRouteFilterPrefixes(assoc.AllowedPrefixesToDirectConnectGateway))
 	if err != nil {
 		return fmt.Errorf("error setting allowed_prefixes: %s", err)
 	}
 
+	if _, ok := d.GetOk("vpn_gateway_id"); ok {
+		d.Set("vpn_gateway_id", assoc.VirtualGatewayId)
+	} else {
+		d.Set("associated_gateway_id", assoc.AssociatedGateway.Id)
+	}
+	d.Set("associated_gateway_owner_account_id", assoc.AssociatedGateway.OwnerAccount)
+	d.Set("associated_gateway_type", assoc.AssociatedGateway.Type)
+	d.Set("dx_gateway_association_id", assoc.AssociationId)
+	d.Set("dx_gateway_id", assoc.DirectConnectGatewayId)
+	d.Set("dx_gateway_owner_account_id", assoc.DirectConnectGatewayOwnerAccount)
+
+	tgwAttachmentId := ""
+	if aws.StringValue(assoc.AssociatedGateway.Type) == directconnect.GatewayTypeTransitGateway {
```

Comment:
We need to get the transit gateway attachment here to ensure that we wait for it's deletion when this resource is deleted.
See discussion: https://github.com/terraform-providers/terraform-provider-aws/issues/8490#issuecomment-490980638.
Maybe move this code plus `ec2TransitGatewayAttachmentStateRefresh()` and `waitForEc2TransitGatewayAttachmentDeletion()` below to `ec2_transit_gateway.go`?